extensions/wikia/DesignSystem/services/DesignSystemGlobalNavigationService.class.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class DesignSystemGlobalNavigationService extends WikiaService {
+	public function index() {
+		$this->setVal( 'model', $this->getData() );
+	}
+
+	public function linkBranded() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
+	}
+
+	public function links() {
+		$this->response->setValues( [
+			'model' => $this->getVal( 'model' ),
+			'type' => $this->getVal( 'type' ),
+			'dropdownRightAligned' => $this->request->getBool( 'dropdownRightAligned' ),
+		] );
+	}
+
+	public function accountNavigation() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
+	}
+
+	public function logo() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
+	}
+
+	public function link() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
+	}
+
+	public function linkDropdown() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
+	}
+
+	public function linkAuthentication() {
+		$model = $this->getVal( 'model' );
+		$messageKey = $model['title']['key'];
+		$href = $model['href'];
+		$classNames = '';
+
+		switch ( $messageKey ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)